lib/btree.js

```diff
@@ -0,0 +1,134 @@
+'use strict';
+
+const DEFAULT_DEGREE = 6; // min degree of b-tree.
+// All vertices except the root have [degree ... 2 * degree] child nodes
+// And [degree - 1 ... 2 * degree - 1] + 1 "empty" elements
+
+class Element {
+  constructor(value, tuple, child = null) {
+    this.value = value;
+    this.tuple = tuple;
+    this.child = child;
```

Comment:
:thinking: it may even be `children`, wdyt?

Reply:
I don't think that it will be more correct, we have here only a one ref to one child and not to several

Reply:
No, this is not a bug. This function split one child into two. The left half will become the child of the new element that will be inserted before element with position ``index`` in the node. And right half remains a child of its element